extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+class PortableInfoboxDataService {
+
+	const IMAGE_FIELD_TYPE = 'image';
+
+	public function getInfoboxDataByTitle( $title ) {
+		$data = [];
+		if ( $title && $title->exists() ) {
+			$data = Article::newFromTitle( $title, RequestContext::getMain() )
+				//on empty parser cache this should be regenerated, see WikiPage.php:2996
+				->getParserOutput()
+				->getProperty( PortableInfoboxParserTagController::INFOBOXES_PROPERTY_NAME );
+		}
+		return $data;
+	}
+
+	public function getImageListFromInfoboxData( $data ) {
```

Comment:
Please describe the param in more detail; Also, maybe calling it infoboxesData would be more clearly showing that it's in fact an array